modules/openy_gc_auth/modules/openy_gc_auth_custom/src/Form/VirtualYCustomLoginForm.php

```diff
@@ -218,15 +230,16 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
       ->loadByProperties(['mail' => $mail]);
     $user = reset($users);
 
-    if ($provider_config->get('enable_email_verification')) {
-      if ($provider_config->get('require_email_verification') || !$user->isActive()) {
-        // Send email verification if user is blocked or if enabled required
-        // email verification.
-        $this->sendEmailVerification($user, $provider_config, $mail);
-        $form_state->setValue('verified', TRUE);
-        $form_state->setRebuild(TRUE);
-        return;
-      }
+    if (($user instanceof User) &&
+      $provider_config->get('enable_email_verification') &&
+      (!$user->isActive() || $this->isVerificationNeeded($user))
```

Comment:
I think, in this place should be
```suggestion
      (!$this->isVerificationNeeded($user))
```

Reply:
Not sure
We should check the user is not blocked -- and request verification otherwise.
But we can do this check inside the `isVerificationNeeded` function, sure